### PR TITLE
REF/BUG: Add ReplyBase and readCommand and refactor accordingly

### PR DESCRIPTION
**Pull request by ViciousDoormat**
Also bitshift by 1 bugfix and some other refactorings

### CODE REVIEW

File: src/EPS.cpp [hunk 1]
```diff
@@ -114,7 +123,7 @@ EPS::config_reply EPS::set_config_params(DWire &wire, uint8_t i2c_address, uint8
     }
 
     // Get the expected length of PAR_VAL for the given PAR_ID
-    uint8_t par_val_length = get_param_length(par_id);
+    uint16_t par_val_length = get_param_length(par_id);
```
Comment (DavidGhergut) [inline src/EPS.cpp]: @ViciousDoormat Can I ask why did you change the data type into `uint16_t`?

**Comment by DavidGhergut** [comment]
Other than the comment above I think all the additions are in order! Great job @ViciousDoormat💪!

File: src/EPS.cpp (hunk 1 above)
Comment (ViciousDoormat) [reply src/EPS.cpp]: I can give a better reply when I've acces to my laptop/PC, but I won't for some time. So I'll do my best from the top of my head.
The documentation of the command requires the parameter to be a uint16 and describes the length as 16 bytes iirc.  Also the corresponding header in the .h file uses a uint16. 
But maybe I understood it incorrectly.

Comment (mavaneerden) [reply src/EPS.cpp]: This is the parameter length though, not the parameter itself. Are there parameters larger than 255?

File: src/EPS.cpp [hunk 2]
```diff
@@ -10,26 +11,26 @@ enum Identifiers {
...
+    GET_PCU_HOUSEKEEPING_DATA_RAW = 0x70,    // reference: page 58 of 87 (ICD)
+    SWITCH_NOMINAL_MODE = 0x30,              // reference: page 38 of 87 (ICD)
+    OUTPUT_BUS_CHANNEL_OFF = 0x18,           // reference: page 37 of 87 (ICD)
+    OUTPUT_BUS_GROUP_STATE = 0x14,           // reference: page 35 of 87 (ICD)
+    OUTPUT_BUS_GROUP_OFF = 0x12,             // reference: page 34 of 87 (ICD)
+    OUTPUT_BUS_GROUP_ON = 0x10,              // reference: page 33 of 87 (ICD)
+    RESET_CONFIGURATION = 0x90               // reference: page 66 of 87 (ICD)
 };
 
 // Enum for reset keys or confirmation keys
```
Comment (mavaneerden) [inline src/EPS.cpp:36]: Make the parameters enums too, then you don't have to validate them

File: src/EPS.h [hunk 3]
```diff
@@ -34,72 +34,50 @@
 
 class EPS {
```
Comment (mavaneerden) [inline src/EPS.h:35]: Make DWire and i2c_address class members so you don't have to put them in every function parameter

File: src/EPS.cpp (hunk 1 above)
Comment (ViciousDoormat) [reply src/EPS.cpp]: image
These are the possible parameter types. Maybe some method to dynamically assign a type, or use something like a Sum type if it exists in C++ would be appropriate here

File: src/EPS.cpp (hunk 2 above)
Comment (ViciousDoormat) [reply src/EPS.cpp:36]: I do agree with this idea and we will implement it. However, this was already mentioned here: https://github.com/davincisatellite/msp-flightsoftware/pull/38#discussion_r1808942096 and is not marked as resolved, so mentioning it again is not necessary.

File: src/EPS.h (hunk 3 above)
Comment (ViciousDoormat) [reply src/EPS.h:35]: I don't think this is a big problem, and changing it would be quite some effort I think as every function is static
